interfaces/machine_parameters/ring_section.py

```diff
@@ -177,98 +168,162 @@ def __init__(self, section_length, alpha_0,
             synchronous_data \
-                = dTypes._ring_program.conversions[func_type](synchronous_data)
+                = getattr(ring_programs, func_type + '_program')(
+                    synchronous_data)
 
         self.synchronous_data = synchronous_data
 
+        # Setting orbit length
+        if orbit_bump is None:
+
+            self.length = ring_programs.orbit_length_program(
+                self.length_design)
+            self.orbit_bump = None
+
+        else:
+
+            if not isinstance(orbit_bump, ring_programs.orbit_length_program):
+                orbit_bump = ring_programs.orbit_length_program(orbit_bump)
+
+            self._check_and_set_alpha_and_orbit(orbit_bump)
+
+            # orbit_bump was checked
+            # The length attribute is created with no further checks
+            if orbit_bump.timebase == 'by_time':
+
+                self.length = ring_programs.orbit_length_program(
+                    [self.length_design] * len(self.orbit_bump[0, 0, :]),
+                    time=self.orbit_bump[0, 0, :])
+                self.length[:, 1, :] += self.orbit_bump[:, 1, :]
+
+            else:
+
+                self.length = self.orbit_bump + self.length_design
+
         # Setting the linear momentum compaction factor
         # Checking that the synchronous data and the momentum compaction
         # have the same length if defined turn-by-turn, raise a warning if one
         # is defined by turn and the other time based
-        self.alpha_order = 0
-        if not isinstance(alpha_0, dTypes.momentum_compaction):
-            alpha_0 = dTypes.momentum_compaction(alpha_0, order=0)
+        self.alpha_orders = [0]
+        alpha_order_max = 0
+        if not isinstance(alpha_0, ring_programs.momentum_compaction):
+            alpha_0 = ring_programs.momentum_compaction(alpha_0, order=0)
         else:
             if alpha_0.order != 0:
                 raise excpt.InputError(
                     "The order of the datatype passed as keyword " +
                     "argument alpha_%s do not match" % (0))
 
-        self._check_and_set_momentum_compaction(alpha_0, 0)
+        self._check_and_set_alpha_and_orbit(alpha_0, 0)
 
         # Treating non-linear momentum compaction factor if declared
         # Listing all the declared alpha first
-        alpha_n = {1: alpha_1, 2: alpha_2}
+        alpha_n = {}
 
         if alpha_1 is not None:
-            self.alpha_order = 1
-        if alpha_2 is not None:
-            self.alpha_order = 2
+            alpha_n[1] = alpha_1
+            alpha_order_max = 1
 
         for argument in kwargs:
             if 'alpha' in argument:
                 try:
                     order = int(argument.split('_')[-1])
-                    self.alpha_order = np.max([self.alpha_order, order])
+                    alpha_order_max = np.max([alpha_order_max, order])
                     alpha_n[order] = kwargs[argument]
```

Comment:
As with the one below, better to put these two outside the `try/except`, an `else` block would work but is not necessary.

Reply:
Included with the above